### DISCUSSION TITLE
Converting JPX with CMYK color profile to RGB

### DISCUSSION THREAD

**Discussion by RobertPennefather** [ImageMagick/ImageMagick#6882] [Help] [answered]
### ImageMagick version

7.1.1-15

### Operating system, version and so on

macOS Ventura 13.6.1 

### Description

Hi there,

I'm trying to convert a JPX with CMYK color space and profile to use RGB while retaining the same colors. I've tried several combinations of command line arguments but can't seem to figure out the right ones. I'm happy if the final output isn't a JPX file just as long as the colors are retained

Here's an example file [cmyk_man.jpx.zip](https://github.com/ImageMagick/ImageMagick/files/13455451/cmyk_man.jpx.zip) with screenshots of some of the output attempts

| Original  | `convert input.jpx output.jpx` | `convert input.jpx -set colorspace CMYK output.jpx` | `convert input.jpx -set colorspace CMYK output.jpg`
| ------------- | ------------- | ------------- | ------------- |
| <img width="646" alt="Screenshot 2023-11-24 at 4 39 08 pm" src="https://github.com/ImageMagick/ImageMagick/assets/17919362/97d9c718-96be-45cd-bece-872c12a7c2fa"> | <img width="649" alt="Screenshot 2023-11-24 at 4 41 39 pm" src="https://github.com/ImageMagick/ImageMagick/assets/17919362/682c37cf-39c3-4e77-abac-bcf8a19e52cf"> | <img width="646" alt="Screenshot 2023-11-24 at 4 39 51 pm" src="https://github.com/ImageMagick/ImageMagick/assets/17919362/d55718af-23f9-49be-ad85-fa151aa1b781"> | <img width="647" alt="Screenshot 2023-11-24 at 4 44 15 pm" src="https://github.com/ImageMagick/ImageMagick/assets/17919362/1fb52ea2-ed96-4e51-a37c-a6fcb987cc2b">

**Comment by snibgo** [comment]
The link is broken. It says "The transfer you requested has been deleted."

**Reply by dlemstra** [reply]
You can upload the file as a zip in this issue.

**Reply by RobertPennefather** [reply]
Sorry, here's the file, updated the original link too [cmyk_man.jpx.zip](https://github.com/ImageMagick/ImageMagick/files/13455478/cmyk_man.jpx.zip)

**Comment by rich2005** [comment]
Using ImageMagick 7.1.1-19  (linux appimage)

magick cmyk_man.jpx  -channel RGB -negate out.jpg

**Comment by snibgo** [comment, marked as answer]
IM seems to be misreading the file.

`exiftool cmyk_man.jpx` says:
```
Number Of Components            : 4
Bits Per Component              : 8 Bits, Unsigned
Compression                     : JPEG 2000
Color Spec Method               : Enumerated
Color Spec Precedence           : 2
Color Spec Approximation        : Accurate
Color Space                     : CMYK
```
`magick cmyk_man.jpx -verbose info` says the image has RGB chanels, and a meta channel. It can't see an ICC color profile.

Gimp can't read the file (a plugin crashes). XnView MP misreads the file as RGB plus alpha.

Maybe the file itself has a problem.

**Reply by RobertPennefather** [reply]
Ah, thanks! I've tried a few other CMYK JPX examples (where the file isn't broken) and `convert input.jpx -set colorspace CMYK output.jpg` does the trick
